Gruntfile.js

```diff
@@ -40,6 +40,24 @@ var getFromExec = function (command, options) {
     return result;
 };
 
+/**
+ * Rename function for grunt file tasks for  adding ".min" convention in front
+ * of minified files; won't do anything to files that already have ".min" in
+ * their filename
+ * @param {String} dest - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} src - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} extension - the extension of the file in question ("js" or "css")
+*/
+var addMinifyToFilename = function (dest, src, extension) {
+    var minifiedExtension = ".min." + extension;
+    // Don't operate on files that already have a .min convention
+    if (src.indexOf(minifiedExtension) > -1) {
+        return dest + src;
+    } else {
+        return dest + src.replace("." + extension, minifiedExtension);
+    }
+};
```

Comment:
It looks like `path.extname(path)` "returns the extension of the path, from the last occurrence of the . (period) character to end of string in the last portion of the path".

This doesn't work with the conventions of generating source maps in a structure like ` infusion-all.js.map`, where the expected output from that would be `infusion-all.min.js.map` to match the source file `infusion-all.min.js`. Using `path.extname(path)` results in improperly placing the `.min` before the `.map`.

I'll noodle on this further, as I think this might be resolveable without a renaming function entirely if `allBuildName` was aware of the `compress` setting. Looking further at the logic of the code, it only currently gets called for dealing with the `all.min` build files to rename them to the minified convention, because the others already have `min` in their filename from their distribution keys. However, that's a side effect rather than intentional, which is dangerous.

It might make more sense to assume all distributions should produce both a non-minified and a minified build, and restructure the logic accordingly to remove the need to define minified distributions in the `distributions` object accordingly.